File: PS/Day8-The Bad Apple/The_Bad_Apple.py
t=int(input())
while t>0:
 a,b=map(int,input().split());c,d=map(int,input().split())
 r=a-c if a>b else b-d
 if r<b-d:r=b-d
 if r<a-c:r=a-c
 if r<d-1:r=d-1
 if r<c-1:r=c-1
 print(r);t-=1

File: PS/Day8-The Bad Apple/The_Bad_Apple.cs
using System;
class s {
    static void Main() {
        int t = int.Parse(Console.ReadLine());
        int
        d = 0;
        while (t > 0) {
            int[] v = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
            int[] m = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
            if (v[0] > v[1]) {
                d = v[0] - m[0];
            } else {
                d = v[1] - m[1];
            }
            if (d <
                v[1] - m[1]) d = v[1] - m[1];
            if (d < v[0] - m[0]) d = v[0] - m[0];
            if (d < m[1] - 1) d = m[1] - 1;
            if (d < m[0] - 1) d = m[0] - 1;
            Console.WriteLine(d);
            t--;
        }
    }
}